### PR TITLE
ar71xx: add support to TP-Link Archer C59v1 and C60v1

### PR DESCRIPTION
**Pull request by heinzek**
TP-Link Archer C59v1 is a dual-band AC1350 router, based on Qualcomm/Atheros
QCA9561+QCA9886.

Specification:
- 775/650/258 MHz (CPU/DDR/AHB)
- 128 MB of RAM (DDR2)
- 16 MB of FLASH (SPI NOR)
- 3T3R 2.4 GHz
- 2T2R 5 GHz
- 5x 10/100 Mbps Ethernet
- USB 2.0 port
- 8x LED (controled by 74HC595), 3x button
- UART header on PCB

TP-Link Archer C60v1 is a dual-band AC1350 router, based on Qualcomm/Atheros
QCA9561+QCA9886.

Specification:
- 775/650/258 MHz (CPU/DDR/AHB)
- 64 MB of RAM (DDR2)
- 8 MB of FLASH (SPI NOR)
- 3T3R 2.4 GHz
- 2T2R 5 GHz
- 5x 10/100 Mbps Ethernet
- 7x LED, 2x button
- UART header on PCB

Currently not working:
- Port LAN1 on C59, LAN4 on C60
- WiFi 5GHz (missing ath10k firmware for QCA9886 chip)
- Update from oficial web interface ( tplink-saveloader not support "product-info")

Flash instruction:
1. Set PC to fixed ip address 192.168.0.66
2. Download lede-ar71xx-generic-archer-cXX-v1-squashfs-factory.bin
and rename it to tp_recovery.bin
3. Start a tftp server with the file tp_recovery.bin in its root directory
4. Turn off the router
5. Press and hold Reset button
6. Turn on router with the reset button pressed and wait ~15 seconds
7. Release the reset button and after a short time
the firmware should be transferred from the tftp server
8. Wait ~30 second to complete recovery.

Flash instruction under U-Boot, using UART:
1. tftp 0x81000000 lede-ar71xx-...-sysupgrade.bin
2. erase 0x9f020000 +$filesize
3. cp.b $fileaddr 0x9f020000 $filesize
4. reset

Add and fix some missing regs to QCA956X.

### CODE REVIEW

File: target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt [hunk 1]
```diff
@@ -1835,3 +1835,24 @@ config ATH79_MACH_TL_WPA8630
...
+
+config ATH79_MACH_ARCHER_C60_V1
+	bool "TP-LINK Archer C60 v1 support"
+	select SOC_QCA956X
+	select ATH79_DEV_AP9X_PCI if PCI
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_WMAC
```
Comment (blogic) [inline target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt]: this does not like it is ordered alphabetically

**Comment by blogic** [comment]
looks good, once the ordering is fixed i can merge this

File: target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt (hunk 1 above)
Comment (heinzek) [inline target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt]: Fixed it with new commit

**Comment by heinzek** [comment]
I sort all board configs in Kconfig.openwrt file.

**Comment by heinzek** [comment]
@blogic its something wrong with this pr?

**Comment by jow-** [comment]
Staged into my tree, thank you!

**Pull request closed without merging**